interpreter/llvm-project/clang/lib/Sema/SemaTemplate.cpp

```diff
@@ -39,31 +39,6 @@
 using namespace clang;
 using namespace sema;
 
-bool HackForDefaultTemplateArg::AllowNonCanonicalSubstEnabled = true;
-
-// Default constructor, record the current value
-// of HackForDefaultTemplateArg::AllowNonCanonicalSubst
-// and set it to true.
-HackForDefaultTemplateArg::HackForDefaultTemplateArg() :
-  OldValue(HackForDefaultTemplateArg::AllowNonCanonicalSubstEnabled)
-{
-  AllowNonCanonicalSubstEnabled = true;
-}
```

Comment:
Re-reading this hack, doesn't this mean `AllowNonCanonicalSubstEnabled` is always `true`, ie the hack was *always* enabled as opposed to "only with the RAII"?

Reply:
Uh, yeah...